src/CalcViewModel/GraphingCalculator/GraphingSettingsViewModel.h

```diff
@@ -10,12 +10,14 @@ namespace CalculatorApp::ViewModel
     [Windows::UI::Xaml::Data::Bindable] public ref class GraphingSettingsViewModel sealed : public Windows::UI::Xaml::Data::INotifyPropertyChanged
     {
     public:
-        OBSERVABLE_OBJECT();
+        OBSERVABLE_OBJECT_CALLBACK(OnPropertyChanged);
```

Comment:
Classes which implement INotifyPropertyChanged shouldn't listen for their own property changes. It's inefficient, and the "stringly-typed" API is prone to break if someone renames properties in the future.

Instead, we should call GraphThemeSettingChanged in the _setter_ for the two properties you've added. We don't have a macro in Utils.h which makes this easy but you could add one along these lines:

```
// The callback specified in the macro is a method in the class that will be called every time the object changes
// The callback takes one argument, which is the new value of the property
#define OBSERVABLE_PROPERTY_RW_CALLBACK(t, n, c)                                                                                                                           \
    property t n                                                                                                                                               \
    {                                                                                                                                                          \
        t get()                                                                                                                                                \
        {                                                                                                                                                      \
            return m_##n;                                                                                                                                      \
        }                                                                                                                                                      \
        void set(t value)                                                                                                                                      \
        {                                                                                                                                                      \
            if (m_##n != value)                                                                                                                                \
            {                                                                                                                                                  \
                m_##n = value;                                                                                                                                 \
                c(value);
                RaisePropertyChanged(L#n);                                                                                                                     \
            }                                                                                                                                                  \
        }                                                                                                                                                      \
    }                                                                                                                                                          \
                                                                                                                                                               \
private:                                                                                                                                                       \
    t m_##n;                                                                                                                                                   \
                                                                                                                                                               \
public:
```